### PR TITLE
Create Binary_Search_Tree.kt

### LINKED ISSUE #2199: Binary Search Trees
- [x] C
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go
- [x] Java
- [x] JS
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by PrachieNaik**
Fixes #2199

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Languages Used:
Kotlin

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by PrachieNaik** [comment]
@asha15 Can you please check now? And if there is problem please suggest me line number

File: Binary_Search_Trees/Binary_Search_Tree.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		val temp = node(value)
+		if (head == null)
+			head = temp
+		else {
+			var current: node?
+			current = head
+			while (current != null) {
+				if (value < current.data) {
+					if (current.left != null)
+						current = current.left
```
Comment (asha15) [inline Binary_Search_Trees/Binary_Search_Tree.kt:20]: please keep four spaces, don't use tab. Do this correction in your entire code

Comment (PrachieNaik) [reply Binary_Search_Trees/Binary_Search_Tree.kt]: Can you please check now? Thank you

File: Binary_Search_Trees/Binary_Search_Tree.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
+class BinarySearchTree { 
+    private var head: node? = null
+
+    class node(var data: Int) { 
+        var left: node? = null
+	var right: node? = null
```
Comment (asha15) [inline Binary_Search_Trees/Binary_Search_Tree.kt:6]: correct the indentation

File: Binary_Search_Trees/Binary_Search_Tree.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    fun Insert(value: Int) {
+	val temp = node(value)
+	if (head == null)
+	    head = temp
+	else {
+	    var current: node?
+	    current = head
+	    while (current != null) {
+	        if (value < current.data) {
```
Comment (asha15) [inline Binary_Search_Trees/Binary_Search_Tree.kt:17]: please correct the indentation

File: Binary_Search_Trees/Binary_Search_Tree.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    fun Search(value: Int) {
+        var current: node?
+        current = head
+        while (current != null) {
+	    current = if (value < current.data) current.left else if (value > current.data) current.right else {
+	        println("Element $value Found")
+	        return
+	    }
```
Comment (asha15) [inline Binary_Search_Trees/Binary_Search_Tree.kt:43]: indentation

File: Binary_Search_Trees/Binary_Search_Tree.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        println("Element $value not Found")
+    }
+
+    fun Min_Value(head: node?): Int {
+        var temp = head
+        while (temp!!.left != null)
+	    temp = temp.left
```
Comment (asha15) [inline Binary_Search_Trees/Binary_Search_Tree.kt:51]: indentation

File: Binary_Search_Trees/Binary_Search_Tree.kt [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	    head.left = Delete_Key(head.left, value)
+        else if (value > head.data)
+	    head.right = Delete_Key(head.right, value)
+        else { 
+	    if (head.left == null)
+	       return head.right
+	    else if (head.right == null)
+	       return head.left
+	    head.data = Min_Value(head.right)
+	    head.right = Delete_Key(head.right, head.data)
```
Comment (asha15) [inline Binary_Search_Trees/Binary_Search_Tree.kt:68]: indentation

**Comment by somya-kapoor** [comment]
Are you working on it?

File: Binary_Search_Trees/Binary_Search_Tree.kt (hunk 6 above)
Comment (PrachieNaik) [reply Binary_Search_Trees/Binary_Search_Tree.kt:68]: What's not right here?

File: Binary_Search_Trees/Binary_Search_Tree.kt (hunk 3 above)
Comment (PrachieNaik) [reply Binary_Search_Trees/Binary_Search_Tree.kt:17]: what's not right here?

**Comment by PrachieNaik** [comment]
In my device identation looks just fine. Do you have any other tool or way to make it correct? Can someone help me with this? @asha15 @jainaman224

**Comment by somya-kapoor** [comment]
Please use 4 space indentation.

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by PrachieNaik** [comment]
> Are you working on it?

YES

**Comment by PrachieNaik** [comment]
@asha15 Please check

**Comment by asha15** [comment]
please squash your commits and push it

**Comment by PrachieNaik** [comment]
@jainaman224 I can't squash it now, can you please use squash and merge option? Or I can raise another pr

**Comment by PrachieNaik** [comment]
> @jainaman224 I can't squash it now, can you please use squash and merge option? Or I can raise another pr

@jainaman224 can you please merge it?

**Comment by PrachieNaik** [comment]
> Please squash your commits or raise a new PR. But do it before tomorrow.

raising another pr

**Pull request closed without merging**